enclave-runner/src/usercalls/mod.rs

```diff
@@ -1083,6 +1092,23 @@ impl<T: UsercallExtension> From<T> for Box<dyn UsercallExtension> {
 struct UsercallExtensionDefault;
 impl UsercallExtension for UsercallExtensionDefault {}
 
+macro_rules! with_exit_handler {
```

Comment:
This seems complicated, I'd much rather have something where we make use of tokio's task cancellation mechanisms such as https://docs.rs/tokio/0.2.13/tokio/runtime/struct.Runtime.html#method.shutdown_timeout

Reply:
Also, if this is wrapping all usercalls, couldn't it be done one level down in the stack (in usercall dispatch)?

Reply:
> use of tokio's task cancellation mechanisms such as https://docs.rs/tokio/0.2.13/tokio/runtime/struct.Runtime.html#method.shutdown_timeout

That would require upgrading to tokio 0.2, I'll explore that option using PR #210

> couldn't it be done one level down in the stack (in usercall dispatch)?

Good idea, I'll give that a shot